sdks/python/apache_beam/transforms/display_test.py

```diff
@@ -137,14 +124,39 @@ def display_data(self):
 
     nspace = '{}.{}'.format(fn.__module__, fn.__class__.__name__)
     expected_items = [
-        ItemMatcher.matches_kvn('complex_url', 'github.com', nspace),
-        ItemMatcher.matches_kvn('my_dd', now, nspace),
-        ItemMatcher.matches_kvn('python_class', HasDisplayData, nspace),
-        ItemMatcher.matches_kvn('static_integer', 120, nspace),
-        ItemMatcher.matches_kvn('static_string', 'static me!', nspace)]
+        DisplayDataItemMatcher.matches_kvn('complex_url',
+                                           'github.com',
+                                           nspace),
+        DisplayDataItemMatcher.matches_kvn('my_dd',
+                                           now,
+                                           nspace),
+        DisplayDataItemMatcher.matches_kvn('python_class',
+                                           HasDisplayData,
+                                           nspace),
+        DisplayDataItemMatcher.matches_kvn('static_integer',
+                                           120,
+                                           nspace),
+        DisplayDataItemMatcher.matches_kvn('static_string',
+                                           'static me!',
+                                           nspace)]
 
     hc.assert_that(dd.items, hc.contains_inanyorder(*expected_items))
 
+  def test_drop_if_none(self):
+    class MyDoFn(beam.DoFn):
+      def display_data(self):
+        return {'some_val': DisplayDataItem('something').drop_if_none(),
+                'non_val': DisplayDataItem(None).drop_if_none(),
+                'def_val': DisplayDataItem(True).drop_if_default(True),
+                'nodef_val': DisplayDataItem(True).drop_if_default(False)}
+
+    dd = DisplayData.create_from(MyDoFn())
+    expected_items = [DisplayDataItemMatcher.matches_kv('some_val',
+                                                        'something'),
+                      DisplayDataItemMatcher.matches_kv('nodef_val',
+                                                        True)]
+    hc.assert_that(dd.items, hc.contains_inanyorder(*expected_items))
```

Comment:
Does this validate that the `dd.items` contains *only* the specified items?  You might need to add an additional assertion `hc.assert_that(dd.items, hc.not(hc.contains_inanyorder(..`